validator/src/main/java/com/amazon/aoc/validators/LoadBalanceValidator.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.aoc.validators;
+
+import com.amazon.aoc.callers.ICaller;
+import com.amazon.aoc.enums.GenericConstants;
+import com.amazon.aoc.exception.BaseException;
+import com.amazon.aoc.exception.ExceptionCode;
+import com.amazon.aoc.fileconfigs.FileConfig;
+import com.amazon.aoc.helpers.MustacheHelper;
+import com.amazon.aoc.helpers.RetryHelper;
+import com.amazon.aoc.helpers.SortUtils;
+import com.amazon.aoc.models.Context;
+import com.amazon.aoc.models.SampleAppResponse;
+import com.amazon.aoc.models.ValidationConfig;
+import com.amazon.aoc.models.xray.Entity;
+import com.amazon.aoc.services.XRayService;
+import com.amazonaws.services.xray.model.Segment;
+import com.amazonaws.services.xray.model.Trace;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.github.wnameless.json.flattener.JsonFlattener;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class LoadBalanceValidator implements IValidator {
+  private MustacheHelper mustacheHelper = new MustacheHelper();
+  private XRayService xrayService;
+  private ICaller caller;
+  private Context context;
+
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper().setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
+
+  // for unit test
+  public void setXRayService(XRayService xrayService) {
+    this.xrayService = xrayService;
+  }
```

Comment:
Creating a method just for unit testing does not seem right. I think there is probably another way to go about this. Possibly change the Class `init` method to accept an `XRayService` object.

Reply:
this is actually a common pattern in Java https://timkranen.medium.com/using-visiblefortesting-to-test-your-private-functions-c9e511fd1c2a but many people don't like.

I think Bryan is right here. You could overload the init method and create a version that accepts a XrayService as parameter.

Reply:
Hmm does this mean we would have to change the CWMetricValidator as well?  In the CWMetricValidator class, it has a [method](https://github.com/aws-observability/aws-otel-test-framework/blob/35a8e23ee4fb81c5d69b023826132266f9639bdd/validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java#L50) for unit tests called setCloudWatchService().  Also, the init method cannot be overloaded it seems like since it is not present in the interface.  It seems like the other validators use the init() method to only set the parameters needed.

Reply:
In java, can't you define multiple of the same method names as long as they have different parameters?

Reply:
I have now abstracted a class that this one now extends and overoaded the init() method to change the XrayService.